evennia/commands/default/building.py

```diff
@@ -531,22 +537,60 @@ def func(self):
         "Define command"
 
         caller = self.caller
-        if not self.args:
+        if not self.args and 'edit' not in self.switches:
             caller.msg("Usage: @desc [<obj> =] >description>")
             return
 
+        desc_set = False
+        obj = None
+
         if self.rhs:
             # We have an =
+            if 'edit' in self.switches:
+                self.msg("{rYou may specify a description, or use the edit switch, "
+                         "but not both.{n")
+                return
             obj = caller.search(self.lhs)
             if not obj:
                 return
-            desc = self.rhs
-        else:
+            obj.db.desc = self.rhs
+            desc_set = True
+        elif 'edit' not in self.switches:
             obj = caller.location
-            desc = self.args
-        # storing the description
-        obj.db.desc = desc
-        caller.msg("The description was set on %s." % obj.key)
+            obj.db.desc = self.args
```

Comment:
This seems like a rather hard-to-read way to do it. Why not just an if ... else statement for the editor or non-editor cases?